### ISSUE TITLE
Support for multilanguage and installation path with different languages

### ISSUE THREAD

**Issue by Purple-CSGO** [advancedfx/advancedfx#198]
Now hlae does not support multi language well enough. Errors occurs if hlae is installed in a path with other languages. It really confuses some new hlae users whose native language isn't English.
-- --
I know a man who has worked on this and succeeded in translating&replacing strings that appear in GUI and Error messages in console. However, it requires lots of work and repeated work after each time a new version is released.
-- --
I don't know how to work with it. Maybe adding support for .xml files? Add a `language` folder then hlae program reads different xml files  e.g. `hlae/language/en_US.xml`  `./zh_CN.xml` etc to display different languages, and then add `language` in main GUI?
-- --
That's only a suggestion. It will not take too long since this man has translated a lot of words. Maybe after some time of varification, these xml files can be added officially into the project.
-- --
Edit by @dtugend:
- [x] Make Installer to solve the path non-latin1 characters problem
- [x] Make Installer translatable
- [x] Make C# parts of HLAE GUI translatable
- [ ] Make C++/CLI parts translatable or port them **^1**
- [ ] Make AfxHookSource translatable **^1**
- [ ] Make AfxHookGoldSrc translatable **^1**

**^1** depends on #317

**Comment by dtugend** [comment]
The problem with the install location can only be solved with an installer. While HLAE has an own bug related to that, the main problem is that these folders can not be mounted into CS:GO, see #65.

I will look into multi language support.
Not sure if it will be XML files, might also be gettext based or similar.

**Comment by dtugend** [comment]
So far only the new installer can be localized:
* https://github.com/advancedfx/advancedfx/tree/main/installer/HlaeCore/lang
* https://github.com/advancedfx/advancedfx/tree/main/installer/setup/lang

But if you or someone else already want to contribute a translation, please have a look at it.

**Comment by cherinyy** [comment]
Hi @dtugend,

For Chinese, considering political factors, I hope that `zh-Hans` and `zh-Hant` can be used instead of `zh-CN` and `zh-TW`.

But I don't know how the program will handle the language setting.

What I have noticed so far:

zh-Hans:

> zh-CN
> zh-SG
> zh-Hans-HK
> zh-Hans-TW
> ...

zh-Hant:

> zh-TW
> zh-HK
> zh-Hant-CN
> zh-Hant-SG
> ...

**Comment by dtugend** [comment]
I understand that things are complicated and to be honest I am not that well informed about the situation in Asia.

Those are international programming conventions / standards for naming them internally in the program: https://en.wikipedia.org/wiki/Language_localisation
It think pretty much all of Linux / GNU GetText uses the same internal codes.

It is up to the translators to translate the language codes politically correct into their own locale and I think this was already done here: https://l10n.advancedfx.org/zh-CN/l10n/all-resources/?search=zh-CN&string=279 - it says 中文（简体）which seems to mean Chinese Simplified.

What could be improved though is the labeling on the official support site: https://www.advancedfx.org/support/
Would changing it to 中文（简体）do?
Are there other similar places that can be improved?

I am not sure if changing the locale files and folder names installed on disk is a good idea, especially for the installer, it would make the code sort of awkward, if it works well at all.

**Comment by cherinyy** [comment]
> I understand that things are complicated and to be honest I am not that well informed about the situation in Asia.
> 
> Those are international programming conventions / standards for naming them internally in the program: https://en.wikipedia.org/wiki/Language_localisation
> It think pretty much all of Linux / GNU GetText uses the same internal codes.
> 
> It is up to the translators to translate the language codes politically correct into their own locale and I think this was already done here: https://l10n.advancedfx.org/zh-CN/l10n/all-resources/?search=zh-CN&string=279 - it says 中文（简体）which seems to mean Chinese Simplified.
> 
> What could be improved though is the labeling on the official support site: https://www.advancedfx.org/support/
> Would changing it to 中文（简体）do?
> Are there other similar places that can be improved?
> 
> I am not sure if changing the locale files and folder names installed on disk is a good idea, especially for the installer, it would make the code sort of awkward, if it works well at all.

Yes, `中文（简体）` means `Chinese (Simplified)`.

So, this is the confusing place: `zh-CN` is strictly `Chinese (China)`, Chinese: `中文（中国）`.

`中文（简体）` or `Chinese (Simplified)` corresponds to `zh-Hans`.

In the actual translation process, in order to avoid the influence of political factors, `zh-CN` will be marked as `zh-Hans` for translation.

If correcting this has a negative impact on the operation of the program or increases the workload, then keep the status quo. But I personally think that this is "not rigorous" in translation.

**Comment by dtugend** [comment]
Do you mean the label in English for the hlae.site website is wrong and should be zh-Hans (?) instead or did I misunderstand?
What should it be in zh-CN files?
What does @Purple-CSGO think?

**Comment by Purple-CSGO** [comment]
I'm not tranlating these words `excactly the same`. Just to have the same mean. And I don't think it's worthy to focus on these little things. It's not gonna related to politcal stuff for the current translation, cuz `CN` / `TW` stands for countries or areas. If you still think it's not 100% correct, then go ahead and push your translation

**Comment by Purple-CSGO** [comment]
> Do you mean the label in English for the hlae.site website is wrong and should be zh-Hans (?) instead or did I misunderstand?
> What should it be in zh-CN files?
> What does @Purple-CSGO think?

translation for `hlae.site` is correct, cuz we name it....

**Comment by cherinyy** [comment]
> Do you mean the label in English for the hlae.site website is wrong and should be zh-Hans (?) instead or did I misunderstand?
> What should it be in zh-CN files?
> What does @Purple-CSGO think?

No, I don’t mean that.
Please ignore what I said, this is a complicated question.
For now, just leave it as it is.

I am sorry for the confusion I caused.

**Comment by dtugend** [comment]
Hi. Thank you for your suggestion. The implementation of this feature in a reasonable time-frame is currently not realistic for me. After considering that together with issue #896 I am closing this request as not planned for now, since it would block other things I would like to still get done before that. Sorry for the copy pasta.
